src/parquet/column/test-util.h

```diff
@@ -240,6 +352,126 @@ static void Paginate(const ColumnDescriptor *d,
+      max_byte_array_len);
+}
+
+template<>
+void InitValues(int num_values, vector<FLBA>& values,
+    vector<uint8_t>& buffer) {
+  int flba_length = 12;
+  size_t nbytes = num_values * flba_length;
+  buffer.resize(nbytes);
+  random_fixed_byte_array(num_values, 0, buffer.data(), flba_length,
+      values.data());
+}
+
+template <typename T>
+static void InitDictValues(int num_values, int dict_per_page,
+    vector<T>& values, vector<uint8_t>& buffer) {
+  int repeat_factor = num_values / dict_per_page;
+  InitValues<T>(dict_per_page, values, buffer);
+  // add some repeated values
+  for (int j = 1; j < repeat_factor; ++j) {
+    for (int i = 0; i < dict_per_page; ++i) {
+      values[dict_per_page * j + i] = values[i];
+    }
+  }
+  // computed only dict_per_page * repeat_factor - 1 values < num_values
+  // compute remaining
+  for (int i = dict_per_page * repeat_factor; i < num_values; ++i) {
+    values[i] = values[i - dict_per_page * repeat_factor];
+  }
+}
+
+// Generates plain pages from randomly generated data
+template <typename Type>
+static int MakePages(const ColumnDescriptor *d, int num_pages, int levels_per_page,
+    vector<int16_t>& def_levels, vector<int16_t>& rep_levels,
+    vector<typename Type::c_type>& values, vector<uint8_t>& buffer,
+    vector<shared_ptr<Page> >& pages,
+    Encoding::type encoding = Encoding::PLAIN) {
+  int num_levels = levels_per_page * num_pages;
+  int num_values = 0;
+  uint32_t seed = 0;
+  int16_t zero = 0;
+  int16_t max_def_level = d->max_definition_level();
+  int16_t max_rep_level = d->max_repetition_level();
+  vector<int> values_per_page(num_pages, levels_per_page);
+  // Create definition levels
+  if (max_def_level > 0) {
+    def_levels.resize(num_levels);
+    random_numbers(num_levels, seed, zero, max_def_level, def_levels.data());
+    for (int p = 0; p < num_pages; p++) {
+      int num_values_per_page = 0;
+      for (int i = 0; i < levels_per_page; i++) {
+        if (def_levels[i + p * levels_per_page] == max_def_level) {
+          num_values_per_page++;
+          num_values++;
+        }
+      }
+      values_per_page[p] = num_values_per_page;
+    }
+  } else {
+    num_values = num_levels;
+  }
+  // Create repitition levels
+  if (max_rep_level > 0) {
+    rep_levels.resize(num_levels);
+    random_numbers(num_levels, seed, zero, max_rep_level, rep_levels.data());
+  }
+  // Create values
+  values.resize(num_values);
+  if (encoding == Encoding::PLAIN) {
+    InitValues<typename Type::c_type>(num_values, values, buffer);
+    Paginate<Type>(d, values, def_levels, max_def_level,
+        rep_levels, max_rep_level, levels_per_page, values_per_page, pages);
+  } else if (encoding == Encoding::RLE_DICTIONARY) {
+    // Calls InitValues and repeats the data
+    InitDictValues<typename Type::c_type>(num_values, levels_per_page, values, buffer);
+    vector<uint8_t> rle_indices;
+
+    shared_ptr<DictionaryPage> dict_page = MakeDictPage<Type>(d, values, encoding,
```

Comment:
If you store the encoded indices in a `shared_ptr<Buffer>` object it will be more manageable (end up with `vector<shared_ptr<Buffer> >`. It would be better to migrate most of our test code to use buffers instead of raw `std::vector` for consistency